stratum/pipelines/stratum-tna/stratum_tna.p4

```diff
@@ -108,7 +112,23 @@ parser StratumIngressParser(packet_in pkt,
         stratum_md = {{0, 0}};
         stratum_md.bridged.setValid();
         stratum_md.bridged.ig_port = ig_intr_md.ingress_port;
-        transition parse_ethernet;
+        transition check_ethernet;
+    }
+
+    state check_ethernet {
+        // We use ethernet-like headers to signal the presence of PacketOut
+        // metadata before the actual ethernet frame.
+        ethernet_t tmp = pkt.lookahead<ethernet_t>();
+        transition select(tmp.eth_type) {
+            ETHERTYPE_PACKET_OUT: parse_packet_out;
```

Comment:
@eth0xFEED Why arn't you using the previously parsed ig_intr_md.ingress_port and compare to cpu_port to find out if it is a packet out? You don't need ethertype checking and the padding then.... Thats what I do in my p4 progs... Or am I missing something here?

Reply:
If you use ethertype like this someone could send in a prepared packet on a normal port and create havoc and security nightmares....🤔

Reply:
The PCIe CPU port depends on the Tofino chip model (can 192 or 320). Worse, 192 is also a recirculation port. In Fabric-tna we had to use this workaround, but for this simpler pipeline matching on both can work as long as we don't use 192 otherwise.

Reply:
Ah, ok... :(

Can port metadata be of help here ? (page 16 of https://raw.githubusercontent.com/barefootnetworks/Open-Tofino/master/PUBLIC_Tofino-Native-Arch-Document.pdf). Although this is probably to far out for an example prog as also filling that port metadata table needs to be done beforehand by the controller. Not very P4 standards compliant... :(

Reply:
I haven't tried port metadata yet.
But yes, packetio gets very tricky on TNA.